pkg/kubectl/cmd/attach.go

```diff
@@ -128,6 +131,9 @@ func (p *AttachOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, argsIn [
 		return cmdutil.UsageError(cmd, fmt.Sprintf("expected fewer than three arguments: POD or TYPE/NAME or TYPE NAME, saw %d: %s", len(argsIn), argsIn))
 	}
 
+	if p.GetPodTimeOut == 0 {
```

Comment:
Make it check for `> 0` and the message something like "... must be higher than zero".